python/lsst/ts/observatory/control/auxtel/latiss.py

```diff
@@ -0,0 +1,600 @@
+            imgtype="ENGTEST",
+            exptime=exptime,
+            n=n,
+            filter=filter,
+            grating=grating,
+            linear_stage=linear_stage,
+            group_id=group_id,
+            checkpoint=checkpoint,
+        )
+
+    async def take_imgtype(
+        self,
+        imgtype,
+        exptime,
+        n,
+        filter=None,
+        grating=None,
+        linear_stage=None,
+        group_id=None,
+        checkpoint=None,
+    ):
+        """Take a series of images of the specified image type.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for flats.
+        n : `int`
+            Number of frames to take.
+        filter : `None` or `int` or `str`
+            Filter id or name. If None, do not change the filter.
+        grating : `None` or `int` or `str`
+            Grating id or name.  If None, do not change the grating.
+        linear_stage : `None` or `float`
+            Linear stage position.  If None, do not change the linear stage.
+        checkpoint : `coro`
+            A optional awaitable callback that accepts one string argument
+            that is called before each bias is taken.
+
+        """
+
+        if imgtype not in self.valid_imagetype:
+            raise RuntimeError(
+                f"Invalid imgtype:{imgtype}. Must be one of "
+                f"{self.valid_imagetype!r}"
+            )
+
+        exp_ids = np.zeros(n, dtype=int)
+
+        if group_id is None:
+            self.log.debug("Generating group_id")
+            group_id = self.next_group_id()
+
+        if imgtype == "BIAS" and exptime > 0.0:
+            self.log.warning("Image type is BIAS, ignoring exptime.")
+
+        for i in range(n):
+            tag = f"{imgtype} {i+1:04} - {n:04}"
+
+            if checkpoint is not None:
+                await checkpoint(tag)
+            else:
+                self.log.debug(tag)
+
+            end_readout = await self.take_image(
+                exptime=exptime if imgtype != "BIAS" else 0.0,
+                shutter=imgtype not in ["BIAS", "DARK"],
+                image_type=imgtype,
+                group_id=group_id,
+                filter=filter if i == 0 else None,
+                grating=grating if i == 0 else None,
+                linear_stage=linear_stage if i == 0 else None,
+            )
+
+            # parse out visitID from filename -
+            # (Patrick comment) this is highly annoying
+            _, _, i_prefix, i_suffix = end_readout.imageName.split("_")
+
+            exp_ids[i] = int((i_prefix + i_suffix[1:]))
+
+        return exp_ids
+
+    async def take_image(
+        self,
+        exptime,
+        shutter,
+        image_type,
+        group_id,
+        filter=None,
+        grating=None,
+        linear_stage=None,
+        science=True,
+        guide=False,
+        wfs=False,
+    ):
+        """Set up the spectrograph and take a series of images.
```

Comment:
I think this just takes a single image? At least I don't see a parameter for "number of images".

I *strongly* suggest doing the setup in `take_images` instead, and deleting this method, because:
* It seems very wasteful to take N images with the same setup and call setup before each one of the images. (I think at one time `take_images` could change the setup for each exposure, but that is no longer true, since you have very nicely simplified the API).
* `take_image` isn't needed as a public method since `take_images` can do the same thing.
* Turning `take_image` into private method `_take_image` isn't much use if `take_images` does the setup, because it just calls expose.

Reply:
ok, I'm setting up the spectrograph in the `take_imgtype` method and calling `expose` afterwards. The `take_image` method is marked as deprecated.